packages/txservice/src/monitor/buffer.ts

```diff
@@ -0,0 +1,49 @@
+import { TransactionServiceFailure } from "../error";
+
+import { Transaction } from "./transaction";
+
+/**
+* @classdesc Really basic class: manages a map/buffer like it's an ordered dictionary, essentially.
+*/
+export class TransactionBuffer {
+  // TODO: Maps in TS technically maintain order based on order of insertion - but in order for us to
+  // guarantee that that order will be sequential in terms of nonce, we'd have to enforce that
+  // in set() (in other words, throw if there's an attempt to insert out of order). Is that what
+  // we want to do? If so, orderedKeys() will become obsolete.
+  private buffer: Map<number, Transaction> = new Map();
+
+  public get pending(): Transaction[] {
+    return this.orderedKeys()
+      .map((nonce) => this.get(nonce) ?? null)
+      .filter(tx => tx != null && !tx.didFinish) as Transaction[];
+  }
+
+  public insert(nonce: number, transaction: Transaction, overwrite = false) {
+    // TODO: detect gap in nonce? throw if there's an attempt to insert out of order?
```

Comment:
Don't think that this is necessarily the responsibility of this class. Would be open to adding it if there were a good way to handle it, which seems like there isn't without going through the loop or adding duplicate logic to determine if you should enter a backfill or something weird

Reply:
Yeah, I mean... maybe we could do a debug log, make sure that monitor is picking up on those gaps... but I think we'll know pretty quickly whether or not monitor is doing its job lol